### PR TITLE
If mapping ends on something with a value, resolve the value, value of value, etc.

### PR DESCRIPTION
**Pull request by arscan**
If a FHIR mapping ends on an SHR type that has a value, this will assume the mapping is intended to follow the values of the corresponding es6 classes as far as possible (hopefully until a primitive type).

### CODE REVIEW

**Comment by cmoesel** [comment]
This isn't exactly how it works...  as the FHIR exporter will stop at the first element that "fits" into the target (regardless of whether it is a primitive or not).  For example, if the target is a FHIR `CodeableConcept`, and `Foo.value` is an `shr.core.CodeableConcept`, the FHIR exporter will stop there because it knows `shr.core.CodeableConcept` is mappable to `CodeableConcept`.  It sounds like your code wouldn't stop there because `shr.core.CodeableConcept` is not a primitive and it has a _value_ that is a primitive (`code`).  Am I understanding that correctly?

**Comment by arscan** [comment]
Following the value only occurs after it has been determined that there is no declared FHIR mapping for that object.  Which is fine, I think, assuming we can figure out that direct mapping case.  In your CodeableConcept example, CodeableConcept in our test set happens to generate a profile because it changes FHIR `text` to `displayText`.  `Foo.value.toFHIR()` would be called, below, which renders the FHIR json properly (instead of just doing Foo.value.value.value until a string/number/etc shows up):

```javascript
    /**
     * Serializes an instance of the CodeableConcept class to a FHIR object.
     * The FHIR is expected to be valid against the CodeableConcept FHIR profile, but no validation checks are performed.
     * @returns {object} a FHIR object populated with the data from the element
     */
    toFHIR() {
      let inst = {};
      if (this.coding !== null) {
        inst['coding'] = this.coding.map(f => typeof f.toFHIR === 'function' ? f.toFHIR() : f);
      }
      if (this.displayText !== null) {
        inst['text'] = typeof this.displayText.toFHIR === 'function' ? this.displayText.toFHIR() this.displayText;
     }
     return inst;
    }
```

Does that make sense?  Or am I missing your point?  I can look at adding a test if I am.

File: lib/generateClass.js [hunk 1]
```diff
@@ -393,7 +401,7 @@ function generateAssignmentIfList(card, jsonString, valueString, cw) {
 
 
 // TODO: Refactor.  How to handle arrays in middle of mapping chains?
-function generateAssignmentIfListFHIR(card, fhirElementPath, valueString, cw) {
+function generateFHIRAssigment(card, fhirElementPath, valueString, cw) {
```
[the remark this answers is not in the document]
Comment (arscan) [reply lib/generateClass.js]: Good catch, fixed.

**Comment by cmoesel** [comment]
Sprry for the delayed response, but I think you understood my main point, @arscan.  Thanks.

**Pull request merged**